cmd/apps/of_loki.go

```diff
@@ -0,0 +1,119 @@
+
+package apps
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+
+	"github.com/alexellis/arkade/pkg/k8s"
+
+	"github.com/alexellis/arkade/pkg"
+	"github.com/alexellis/arkade/pkg/apps"
+	"github.com/alexellis/arkade/pkg/config"
+	"github.com/alexellis/arkade/pkg/env"
+	"github.com/alexellis/arkade/pkg/helm"
+	"github.com/alexellis/arkade/pkg/types"
+	"github.com/spf13/cobra"
+)
+
+func MakeInstallOpenFaaSLoki() *cobra.Command {
+	var OpenFaaSlokiApp = &cobra.Command{
+		Use:          "openfaas-loki",
+		Short:        "Install Loki-OpenFaaS and Configure Loki logs provider for OpenFaaS",
+		Long:         "Install Loki-OpenFaaS and Configure Loki logs provider for OpenFaaS",
+		Example:      "arkade install openfaas-loki",
+		SilenceUsage: true,
+	}
+
+	OpenFaaSlokiApp.Flags().StringP("namespace", "n", "default", "The namespace to install loki (default: default")
+	OpenFaaSlokiApp.Flags().Bool("update-repo", true, "Update the helm repo")
+	OpenFaaSlokiApp.Flags().String("openfaas-namespace", "openfaas", "set the namespace that OpenFaaS is installed into")
+	OpenFaaSlokiApp.Flags().String("loki-url", "http://loki-stack.default:3100", "set the loki url (default http://loki-stack.default:3100)")
+	OpenFaaSlokiApp.Flags().StringArray("set", []string{}, "Use custom flags or override existing flags \n(example --set grafana.enabled=true)")
+
+	OpenFaaSlokiApp.RunE = func(command *cobra.Command, args []string) error {
+		helm3 := true
+
+		namespace, _ := OpenFaaSlokiApp.Flags().GetString("namespace")
+		userPath, err := config.InitUserDir()
+		if err != nil {
+			return err
+		}
+
+		clientArch, clientOS := env.GetClientArch()
+
+		log.Printf("Client: %s, %s\n", clientArch, clientOS)
+
+		log.Printf("User dir established as: %s\n", userPath)
+
+		if err := os.Setenv("HELM_HOME", path.Join(userPath, ".helm")); err != nil {
+			return err
+		}
+
+		openfaasNamespace, _ := OpenFaaSlokiApp.Flags().GetString("openfaas-namespace")
+		lokiURL, _ := OpenFaaSlokiApp.Flags().GetString("loki-url")
+
+		overrides := map[string]string{}
+		overrides["lokiURL"] = lokiURL
+
+		customFlags, _ := command.Flags().GetStringArray("set")
+
+		if err := config.MergeFlags(overrides, customFlags); err != nil {
+			return err
+		}
+
+		lokiOptions := types.DefaultInstallOptions().
+			WithNamespace(namespace).
+			WithHelmPath(path.Join(userPath, ".helm")).
+			WithHelmRepo("lucas/openfaas-loki").
+			WithHelmURL("https://lucasroesler.com/openfaas-loki").
+			WithOverrides(overrides)
+
+		if command.Flags().Changed("kubeconfig") {
+			kubeconfigPath, _ := command.Flags().GetString("kubeconfig")
+			lokiOptions.WithKubeconfigPath(kubeconfigPath)
+		}
+
+		os.Setenv("HELM_HOME", path.Join(userPath, ".helm"))
+
+		_, err = helm.TryDownloadHelm(userPath, clientArch, clientOS, helm3)
+		if err != nil {
+			return err
+		}
+
+		_, err = apps.MakeInstallChart(lokiOptions)
+		if err != nil {
+			return err
+		}
+
+		// Post install config of openfaas-loki
```

Comment:
This bit of magic is nice, but also magic. Can you print out what they should do?

And also as an `arkade install openfaas --set` variation? Imagine if they reinstalled openfaas or were using helm / flux, this would not be persisted.

Reply:
Added to the output (what to add to openfaas using arkade flag, or helm)